### PR TITLE
fix : dead code removed or possible bugs fixed post migration php 8.2

### PR DESCRIPTION
**Pull request by Drago37**
### Fixed

* dead code removed or possible bugs fixed post migration php 8.2

**Fixes** # (issue)

## Type of change

- [x] Patch fixing an issue (non-breaking change)
- [ ] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 22.10.x
- [ ] 23.04.x
- [ ] 23.10.x
- [ ] 24.04.x
- [x] develop

<h2> How this pull request can be tested ? </h2>

Check source code with debug if need.

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [x] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [x] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/www/class/centreonUser.class.php [hunk 1]
```diff
@@ -488,7 +480,7 @@ public function getContactParameters($db, $parameters = [])
      * @param CentreonDB $db
      * @param array $parameters
      *
-     * @return null
+     * @return void
```
Comment (a-launois) [inline centreon/www/class/centreonUser.class.php:483]: There is a return null L489

**Review by a-launois — CHANGES_REQUESTED** [review]

File: centreon/www/class/centreon-clapi/centreon.Config.Poller.class.php [hunk 2]
```diff
@@ -205,7 +205,15 @@ public function pollerReload($variables)
...
         $return_code = $this->writeToCentcorePipe($reloadCommand, $host["id"]);
+        if ($return_code === 1) {
+            print "Error while writing the command {$reloadCommand} in centcore pipe file " . PHP_EOL;
+            exit(1);
+        }
         $return_code = $this->writeToCentcorePipe('RELOADBROKER', $host["id"]);// FIXME variable erased
+        if ($return_code === 1) {
+            print "Error while writing the command RELOADBROKER in centcore pipe file " . PHP_EOL;
+            exit(1);
+        }
```
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/www/class/centreon-clapi/centreon.Config.Poller.class.php:216]: I saw it with @callapa to exit on error, @callapa can you confirm it's ok please ? Thanks.
Also, I fix the pollerRestart method 😄

File: centreon/www/class/centreonConnector.class.php [hunk 3]
```diff
@@ -580,43 +578,4 @@ public static function getDefaultValuesParameters($field)
         return $parameters;
     }
 
-    /**
```
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/www/class/centreonConnector.class.php:583]: arf no I am not sure so I have been a rollback of this deleted method and I have fixed a bug with a variable not initialized. Can you see if it's ok please ?

File: centreon/www/class/centreonTraps.class.php [hunk 4]
```diff
@@ -337,9 +337,9 @@ public function duplicate($traps = [], $nbrDup = []): void
...
 
     /**
-     * @param null $traps_id
+     * @param int|null $traps_id
      *
-     * @return null
+     * @return void
      * @throws InvalidArgumentException
      * @throws PDOException
      */
```
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/www/class/centreonTraps.class.php:345]: php docs fixed thanks !

File: centreon/www/class/centreonUser.class.php (hunk 1 above)
Comment (Drago37) [reply centreon/www/class/centreonUser.class.php]: php docs fixed

File: centreon/www/class/centreonUser.class.php [hunk 5]
```diff
@@ -62,18 +62,10 @@ class CentreonUser
...
-    /** @var */
-    public $num; //FIXME not in this class
     /** @var mixed|null */
     public $gmt;
     /** @var bool|null */
     public $is_admin = null;
-    /** @var */
-    public $groupList; //FIXME not in this class
-    /** @var */
-    public $groupListStr; //FIXME not in this class
```
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/www/class/centreonUser.class.php:76]: A hab done a rollback to add the deleted properties. Thanks

File: centreon/www/class/centreon-clapi/centreon.Config.Poller.class.php [hunk 6]
```diff
@@ -292,19 +292,20 @@ public function pollerRestart($variables)
...
+            echo "Error while writing the command RELOADBROKER in centcore pipe file for host id {$host["id"]}" . PHP_EOL;
+            return $return_code;
         }
         $msg_restart = _("OK: A restart signal has been sent to '" . $host["name"] . "'");
-        print $msg_restart . "\n";
+        echo $msg_restart . "\n";
         $statement = $this->DB->prepare(
             "UPDATE `nagios_server` SET `last_restart` = :last_restart, `updated` = '0' WHERE `id` = :poller_id LIMIT 1"
         );
+        // centreon -u admin -p Centreon\!2021 -a POLLERRELOAD -v 1
```
Comment (a-launois) [inline centreon/www/class/centreon-clapi/centreon.Config.Poller.class.php:308]:
```suggestion
```

File: centreon/www/class/centreon-clapi/centreon.Config.Poller.class.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (Drago37) [reply centreon/www/class/centreon-clapi/centreon.Config.Poller.class.php]: fixed

File: centreon/www/class/centreon-clapi/centreon.Config.Poller.class.php (hunk 6 above)
Comment (Drago37) [reply centreon/www/class/centreon-clapi/centreon.Config.Poller.class.php]: fixed

**Review by a-launois — APPROVED** [review]

**Pull request merged**